src/psyclone/dynamo0p3.py

```diff
@@ -8667,6 +8617,83 @@ def ref_name(self, function_space=None):
             "DynKernelArgument.ref_name(fs): Found unsupported argument "
             "type '{0}'.".format(self._argument_type))
 
+    def _init_data_type_properties(self):
+        '''
+        Set up kernel argument information from LFRicConstants.DATA_TYPE_MAP:
+        module name, data type, proxy data type and precision.
+        This is currently supported for real scalars, fields and operators.
+        Integer and logical scalars are not defined as data structures in
+        the LFRic infrastructure so their precision is set from the defaults
+        in the PSyclone configuration file.
+
+        '''
+        const = LFRicConstants()
+        api_config = Config.get().api_conf("dynamo0.3")
+
+        # All supported scalars have the same metadata, GH_SCALAR, so we
+        # check by their intrinsic type
+        if self.is_scalar:
+            if self.intrinsic_type == "real":
+                # Set 'real' scalar properties as defined in the LFRic
+                # infrastructure
+                self._module_name = const.DATA_TYPE_MAP["scalar"]["module"]
+                self._data_type = const.DATA_TYPE_MAP["scalar"]["type"]
+                self._proxy_data_type = const.DATA_TYPE_MAP[
+                    "scalar"]["proxy_type"]
+                self._precision = const.DATA_TYPE_MAP["scalar"]["kind"]
+                # TODO #1277: Check whether the access of a 'real' scalar
+                # is a reduction before assigning precision from the
+                # algorithm layer. LFRic reductions must have 'r_def'
+                # precision as defined for the "scalar_type".
+            elif self.intrinsic_type == "integer":
```

Comment:
I'm not sold on the idea that we declare precision information within LFRicConstants for real scalars because LFRic defines it and we read precision information from a config file for integer and logical scalars because LFRic does not define them. I think that all of the information should either come from one or the other. I think I buy your argument that a config file is not needed as we know the value for reals. However, we also know what the value is going to be in LFRic for integer and logical, even if it is not yet specified, so there is no reason not to add that information directly in LFRicConstants for them too.

Reply:
As discussed, the `_init_data_type_properties` method now differentiates between real scalar reductions and read-only scalars. A simple map between the intrinsic type and precision of for scalars to `LFRicConstants` is used for the latter, with the explicit information about precision in the map.